lib/oedipus_lex.rb

```diff
@@ -62,9 +55,30 @@ def to_ruby state, predicates, exclusive
         when /^:/, "nil" then
           "  [:state, #{action}]"
         else
+          uses_text = true
           "  #{action} text"
         end
 
+      if uses_text or (action and action.include? 'text')
```

Comment:
Part of the problem is that the action can be any arbitrary code, so just looking for `"text"` can result in false positives. At the very least, I'm putting `\b` on both sides.